crates/ruff_linter/src/rules/ruff/rules/ambiguous_unicode_character.rs

```diff
@@ -196,25 +196,29 @@ pub(crate) fn ambiguous_unicode_character_string(checker: &mut Checker, string_l
         match part {
             ast::StringLikePart::String(string_literal) => {
                 let text = checker.locator().slice(string_literal);
+                let mut diagnostics = Vec::new();
                 ambiguous_unicode_character(
-                    &mut checker.diagnostics,
+                    &mut diagnostics,
```

Comment:
This will be slightly less performant because we now allocate a temporary `Vec` vs just writing to the `checker.diagnostics`. I think my preference here is to just pass `&Checker` (and remove the `checker.settings` argument.

Reply:
I must admit, I didn't read the commit messages in detail. Shame on me! 

I think it's fine then. We could do something similar to Red Knot where we have a trimmed down struct for any context that can emit diagnostics but it feels overkill here.